### PR TITLE
[13.0][MIG] account_financial_report: Migration to 13.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Based on this refactoring on module for 12.0 version: https://github.com/OCA/account-financial-reporting/pull/660

Cc @Tecnativa TT19684

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@gurneyalex finally we have a PR. Reports are working, but still some work to do on tests + fine tuning taxes report.

**Comment by ernestotejeda** [comment]
Ready to be reviewed

**Review by HaraldPanten — CHANGES_REQUESTED** [review]
Hi @ernestotejeda I couldn't test the module in runbot, even rebuilding it. But I did that testing in local environment. These are my comments:

1- I wasn't able to download any .xlsx file from the reports. User clicks on "Export XLSX" button but it seems that nothing happens.

2- There's an issue in "Open Items report". You can reproduce it like:

- Go to Accounting module --> Reporting -> "Open items".
- You don't need to change any of the default settings.
- Click to "View button" or "Export PDF" button.

You'll see:

`Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_http.py", line 229, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 804, in dispatch
    r = self._call_function(**self.params)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 345, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/service/model.py", line 93, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 338, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 910, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/auto/addons/report_xlsx/controllers/main.py", line 50, in report_routes
    reportname, docids, converter, **data
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/auto/addons/web/controllers/main.py", line 1956, in report_routes
    html = report.with_context(context).render_qweb_html(docids, data=data)[0]
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_actions_report.py", line 757, in render_qweb_html
    data = self._get_rendering_context(docids, data)
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_actions_report.py", line 774, in _get_rendering_context
    data.update(report_model._get_report_values(docids, data=data))
  File "/opt/odoo/auto/addons/account_financial_report/report/open_items.py", line 344, in _get_report_values
    account_ids, partner_ids, date_at_object, target_move, company_id, date_from
  File "/opt/odoo/auto/addons/account_financial_report/report/open_items.py", line 275, in _get_data
    "date_maturity": move_line["date_maturity"].strftime("%d/%m/%Y"),
AttributeError: 'NoneType' object has no attribute 'strftime'`.

"Export XLSX" button is not failing because no action is done, as I reported in issue number 1.

Could you check please?

Let me know if you need any extra help. Thanks!

**Comment by ernestotejeda** [comment]
Thanks @HaraldPanten , I have made the changes to take into account cases when there are Journal Items without Due date.
Regarding the "Export XLSX" button problem: I couldn't reproduce that locally. According to the tests I have done locally, I can export the XLSX without problems.

File: account_financial_report/report/journal_ledger.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+        """
+
+    def _get_query_taxes_params(self, move_lines):
+        return {"move_line_ids": tuple(move_lines.ids)}
+
+    def _get_move_lines(self, move_ids, wizard, journal_ids):
+        move_lines = self.env["account.move.line"].search(
+            self._get_move_lines_domain(move_ids, wizard, journal_ids),
+            order=self._get_move_lines_order(move_ids, wizard, journal_ids),
+        )
```
Comment (HviorForgeFlow) [inline account_financial_report/report/journal_ledger.py:179]: If move_lines is empty query raises an error:

Back
500: Internal Server Error
Traceback
Traceback (most recent call last):
  File "/home/hveficent/Eficent/odoo/13.0/odoo/addons/base/models/ir_http.py", line 229, in _dispatch
    result = request.dispatch()
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 804, in dispatch
    r = self._call_function(**self.params)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 345, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/service/model.py", line 93, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 338, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 910, in __call__
    return self.method(*args, **kw)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "/home/hveficent/Eficent/odoo/addons/addons-13.0/OCA/reporting-engine/report_xlsx/controllers/main.py", line 49, in report_routes
    reportname, docids, converter, **data
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "/home/hveficent/Eficent/odoo/13.0/addons/web/controllers/main.py", line 1956, in report_routes
    html = report.with_context(context).render_qweb_html(docids, data=data)[0]
  File "/home/hveficent/Eficent/odoo/13.0/odoo/addons/base/models/ir_actions_report.py", line 757, in render_qweb_html
    data = self._get_rendering_context(docids, data)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/addons/base/models/ir_actions_report.py", line 774, in _get_rendering_context
    data.update(report_model._get_report_values(docids, data=data))
  File "/home/hveficent/Eficent/odoo/addons/addons-13.0/OCA/account-financial-reporting/account_financial_report/report/journal_ledger.py", line 305, in _get_report_values
    move_lines = self._get_move_lines(move_ids, wizard, journal_ids)
  File "/home/hveficent/Eficent/odoo/addons/addons-13.0/OCA/account-financial-reporting/account_financial_report/report/journal_ledger.py", line 184, in _get_move_lines
    self.env.cr.execute(query_taxes, query_taxes_params)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/sql_db.py", line 163, in wrapper
    return f(self, *args, **kwargs)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/sql_db.py", line 240, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: syntax error at or near ")"
LINE 7:             WHERE account_move_line_id IN ()
                                                   ^

**Review by HaraldPanten — APPROVED** [review]
Maybe EXCEL failing was an isuue of my local database. Don't worry.

Tested on runbot and it seems to work fine. Functional review 👍 !

File: account_financial_report/report/journal_ledger.py (hunk 1 above)
Comment (ernestotejeda) [reply account_financial_report/report/journal_ledger.py:179]: Thanks @HviorForgeFlow , I see the problem. I add a conditional to validate that case.

**Comment by ernestotejeda** [comment]
Changes done

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍 but......
Don't you think that 'target_move'/'move_target' field should be 'posted' by default?

**Comment by ernestotejeda** [comment]
> Don't you think that 'target_move'/'move_target' field should be 'posted' by default?

Yes, I think so, what do you think @pedrobaeza ?

**Comment by pedrobaeza** [comment]
Yes, please. Now that draft invoices are unposted moves, it has more sense to put that default.

**Comment by ernestotejeda** [comment]
@ValentinVinagre Changes done

**Comment by HaraldPanten** [comment]
Hi @ernestotejeda , could you have a look at Travis?

**Comment by ValentinVinagre** [comment]
I don't know if what I will comment should be done now or could be done in a future improvement ...
When 'all' is selected in the '' target_move '/' move_target 'field, all journal entries are displayed, even cancelled ones. It makes sense in older versions, because journal entries could not be cancelled (only draft or posted). But now, should the cancelled journal entries be shown? 
I think that we only need to show the "draft" or "posted" ones.

What do you think @pedrobaeza @HaraldPanten @ernestotejeda ?

**Comment by pedrobaeza** [comment]
Well, all is all.. and if we have a column indicating that state, for me is enough.

**Comment by ValentinVinagre** [comment]
If it's enough for others, It's OK.

**Comment by HaraldPanten** [comment]
It's not essential, but it's a good idea. I would prefer to purpose that feature for a new IMP PR or open an issue to talk about that.

IMO, this PR can be merged. 🤙 @ValentinVinagre @ernestotejeda @pedrobaeza

**Comment by HaraldPanten** [comment]
@HviorForgeFlow is it ready for you? Thx!

**Review by HviorForgeFlow — APPROVED** [review]

**Comment by HaraldPanten** [comment]
@sbidoul  @JordiBForgeFlow  @pedrobaeza  I think this PR is ready to merge.

Thx!

**Review by manuelcalerosolis — APPROVED** [review]
Code and functional revised

**Comment by HaraldPanten** [comment]
Why did checks fail? They seem to be ok. Aren't them? @pedrobaeza

**Comment by MiquelRForgeFlow** [comment]
Because prettier xml was included in the pre-commit. Thus, this needs to be rebased.

**Comment by HaraldPanten** [comment]
Thanks!

@ernestotejeda Could you rebase?

**Comment by usmanfarzandali** [comment]
why not profit & loss and balance sheet included in OCA reports whats store behind it ?

**Comment by pedrobaeza** [comment]
They are obtained through MIS reports

**Comment by ernestotejeda** [comment]
@pedrobaeza I rebased and I add the changes corresponding to prettier xml in commit [IMP] account_financial_report: black, isort

**Comment by JordiBForgeFlow** [comment]
Thanks to everybody!! @ernestotejeda based on your findings, are we good to merge #660 in v12? Does anything need to be backported from this PR?

**Comment by pedrobaeza** [comment]
I'm a bit more afraid of doing this on a stable version because it will mean a change when they update. Can you assure there's no "visual change" nor possible regressions?

**Comment by ernestotejeda** [comment]
> Thanks to everybody!! @ernestotejeda based on your findings, are we good to merge #660 in v12? Does anything need to be backported from this PR?

@JordiBForgeFlow I saw that you kept #660 updated with the changes that were appearing here and I don't think there is anything to added

**Comment by HaraldPanten** [comment]
Hi @JordiBForgeFlow I think that here we have only the small change that sets "published entries" option by default instead of "all entries" as before. 

It's just a small improvement, but it could be taken in consideration in https://github.com/OCA/account-financial-reporting/pull/660 . What do you think?

**Pull request merged**